src/Dropdown/Dropdown.test.tsx

```diff
@@ -1,213 +1,820 @@
+
+        act(() => {
+            menu.dispatchEvent(new KeyboardEvent("keydown", { bubbles: true, key: "tab", shiftKey: true }));
+        });
+
+        expect(menu.classList.contains("show")).toBeFalsy();
+
+        act(() => {
+            toggle.dispatchEvent(new KeyboardEvent("keydown", { bubbles: true, key: "tab" }));
+        });
+
+        act(() => {
+            toggle.dispatchEvent(new KeyboardEvent("keydown", { bubbles: true, key: "a" }));
+        });
+
+        act(() => {
+            toggle.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+        });
+
+        act(() => {
+            toggle.dispatchEvent(new MouseEvent("mousedown", { bubbles: true }));
+        });
+
+        act(() => {
+            toggle.dispatchEvent(new KeyboardEvent("keydown", { bubbles: true, key: "tab" }));
+        });
+
+        expect(menu.classList.contains("show")).toBeFalsy();
+    });
+
+    it("Should focus back on search if another items is highlighted and the user starts typing", () => {
+        act(() => {
+            render(<Dropdown {...DEFAULT_PROPS} searchable={true} />, container);
+        });
+
+        const toggle: Element = container.querySelector(".custom-dropdown-toggle");
+        expect(toggle).toBeTruthy();
+
+        act(() => {
+            toggle.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+        });
+
+        const searchInput: Element = container.querySelector(".search-input");
+        expect(searchInput).toBeTruthy();
+
+        expect(document.activeElement).toBe(searchInput);
+
+        const menu: Element = container.querySelector(".custom-dropdown-menu");
+        expect(menu).toBeTruthy();
+        expect(menu.classList.contains("show")).toBeTruthy();
+
+        act(() => {
+            menu.dispatchEvent(new KeyboardEvent("keydown", { bubbles: true, key: "arrowdown" }));
+        });
+
+        const options: NodeListOf<Element> = container.querySelectorAll(".custom-dropdown-item:not(.search-input)");
+        expect(options.length).toBe(3);
+        expect(options.item(0).classList.contains("highlighted")).toBeTruthy();
+        expect(document.activeElement).toBe(options.item(0));
+
+        searchInput.setAttribute("value", "A");
+
+        act(() => {
+            searchInput.dispatchEvent(new Event("change", { bubbles: true }));
+        });
+
+        expect(document.activeElement).toBe(searchInput);
+    });
+
+    it("Should display clear button but make it not clickable if disabled and clearable are both set and at least one item is selected", () => {
+        const props: DropdownProps = {
+            ...DEFAULT_PROPS,
+            clearable: true,
+            disabled: true,
+            selectedValue: { ...DEFAULT_PROPS.list[0] }
+        };
+
+        act(() => {
+            render(<Dropdown {...props} />, container);
+        });
+
+        const clearButton: Element = container.querySelector(".right-items > .dropdown-icon-holder:not(.chevron");
```

Comment:
How does this work when you seem to be missing a close bracket ?

Reply:
ohh the closing bracket inside the string was missing, thanks.